### PR TITLE
added license check to Makefiles and CI

### PR DESCRIPTION
**Pull request by TomasLudvik**
#### Description, the reason for the PR
We want to ensure that Shopsys Platform and projects build on it are using dependencies that are allowed for commercial use. This PR ensures that no other licenses are used.

Also improved codeception acceptance tests to ensure empty password validation is always revalidated after password is filled in administration login page.

#### License Agreement for contributions

- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

## Summary by CodeRabbit

- New Features
  - Added automated license compliance checks for PHP and Storefront.
  - Introduced a check-licenses Make target and integrated it into existing quality checks.

- Chores
  - CI pipelines (GitHub/GitLab) now run license checks on PRs and forks.
  - Storefront CI builds from source in dev mode with a mounted workspace.
  - Updated CI Docker images for acceptance-related jobs.

- Dependencies
  - Added license-checker as a development dependency in app and storefront.

- Documentation
  - Added upgrade notes for enabling license checks in Makefile and CI.

----
:globe_with_meridians: Live Preview:
  - https://tl-license-check.odin.shopsys.cloud
  - https://cz.tl-license-check.odin.shopsys.cloud

### CODE REVIEW

**Comment by vitek-rostislav** [comment]
@coderabbitai review

File: .github/workflows/docker-build.yaml [hunk 1]
```diff
@@ -804,6 +804,33 @@ jobs:
...
+                uses: actions/download-artifact@v4
+                with:
+                    name: docker-compose
+            -   name: Build application
+                run: |
+                    docker compose up -d
+                    docker compose exec -T php-fpm composer install --optimize-autoloader --no-interaction --dev
+                    docker compose exec -T php-fpm npm install
+            -   name: Checks dependency licenses in Composer and NPM (php-fpm & storefront)
+                run: make check-licenses
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply .github/workflows/docker-build.yaml]: @TomasLudvik this seems valid

File: .github/workflows/docker-build.yaml [hunk 2]
```diff
@@ -868,6 +895,8 @@ jobs:
                     docker compose exec -T php-fpm php bin/console shopsys:cdn-domain-url:replace
                     docker compose exec -T php-fpm php bin/console shopsys:list:export-scopes
                     docker compose exec -T php-fpm php bin/console shopsys:dispatch:recalculations 1
+            -   name: Check licenses of dependencies in Composer and NPM (php-fpm & storefront)
+                run: make check-licenses
             -   name: PHP-FPM container logs
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply .github/workflows/docker-build.yaml:901]: same here (I would go with option A)

**Comment by vitek-rostislav** [comment]
reviewed and tested locally, disallowed licenses are properly reported :ok_hand: 

image

**Review by vitek-rostislav — APPROVED** [review]

File: project-base/gitlab/docker-compose-ci.yml [hunk 3]
```diff
@@ -28,10 +28,16 @@ services:
             - "8000:8080"
 
     storefront:
-        image: ${STOREFRONT_TAG}
+        build:
+            context: ./storefront
+            dockerfile: ./docker/ci.Dockerfile
```
Comment (grossmannmartin) [inline project-base/gitlab/docker-compose-ci.yml]: This probably prolongs the build as each stage using this docker-compose file now needs to build the image.

Also, I'm not sure if it doesn't break the Cypress screenshots generation as it's now run in dev mode

File: Makefile [hunk 4]
```diff
@@ -204,3 +205,9 @@ generate-snapshots-info-table: ## Generates overview table of Cypress snapshots
 	-docker compose exec storefront-cypress npm run generate-snapshots-table --prefix cypress || true
 	@echo "✅ Snapshots info table generation finished."
 	$(call _cypress-cleanup)
+
+check-licenses: ## Checks dependency licenses in Composer and NPM (php-fpm & storefront)
+	bash -lc 'set -o pipefail; \
```
Comment (grossmannmartin) [inline Makefile]: `set -o pipefail` is superfluous, as there is no pipe in the commands

File: Makefile [hunk 5]
```diff
@@ -204,3 +205,9 @@ generate-snapshots-info-table: ## Generates overview table of Cypress snapshots
...
+	  docker compose exec -T php-fpm bash -lc "project-base/scripts/check-licenses.sh"; s1=$$?; \
```
Comment (grossmannmartin) [inline Makefile]: I would drop -l from every bash/sh call, as I don't think the script should be dependent on any custom profile settings.

File: project-base/.gitlab-ci.yml [hunk 6]
```diff
@@ -98,16 +98,19 @@ test:storefront-standards-with-codegen:
...
     tags:
         - tests
     script:
         - cp -f ./gitlab/docker-compose-ci.yml ./docker-compose.yml
         - docker compose up -d
-        - docker compose exec -T php-fpm /bin/bash -c "mv PRODUCTION DEVELOPMENT; php phing composer-dev npm-install-dependencies test-db-create frontend-api-generate-new-keys tests tests-acceptance-build tests-acceptance"
+        - docker compose exec -T php-fpm /bin/bash -c "mv PRODUCTION DEVELOPMENT; php phing composer-dev npm-install-dependencies test-db-create frontend-api-generate-new-keys"
+        - docker compose cp ./scripts php-fpm:/var/www/html/scripts
+        - docker compose exec -T php-fpm bash -lc "EXTRA_ALLOWED=GPL-2.0-only,GPL-3.0-only,LGPL-2.1-only scripts/check-licenses.sh"
+        - docker compose exec -T storefront sh -lc "EXTRA_ALLOWED=GPL-2.0-only,GPL-3.0-only,LGPL-2.1-only sh scripts/check-licenses.sh"
```
Comment (grossmannmartin) [inline project-base/.gitlab-ci.yml]: `-l` is not needed

File: project-base/storefront/scripts/check-licenses.sh [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+if [ -n "$S_OFF" ]; then
+    echo ""
+    echo "These packages in PNPM use not approved license:"
+    printf "%s\n" "$S_OFF" | sort -u | sed 's/^/- /'
+    echo ""
+    exit 1
+fi
+
+echo ""
+echo "PNPM: All dependencies use approved licenses."
```
Comment (grossmannmartin) [inline project-base/storefront/scripts/check-licenses.sh]: These two scripts scream "AI" from a distance.
It's an overcomplicated solution, and on top of that, really hard to read. 

We need node to run this script, so why don't we write it in JS directly? 
It would be much closer to a common developer. 

license-checker even allows programatic use https://www.npmjs.com/package/license-checker#requiring

File: project-base/gitlab/docker-compose-ci.yml (hunk 3 above)
Comment (TomasLudvik) [reply project-base/gitlab/docker-compose-ci.yml]: Reverted

File: Makefile (hunk 4 above)
Comment (TomasLudvik) [reply Makefile]: Removed

File: Makefile (hunk 5 above)
Comment (TomasLudvik) [reply Makefile]: Done

File: project-base/.gitlab-ci.yml (hunk 6 above)
Comment (TomasLudvik) [reply project-base/.gitlab-ci.yml]: Done

File: project-base/storefront/scripts/check-licenses.sh (hunk 7 above)
Comment (TomasLudvik) [reply project-base/storefront/scripts/check-licenses.sh]: I see this as one time thing, that nobody will change in the future, only might add or remove some license. As it is dependent both on composer and npm (pnpm) I thought solving this via bash will be sufficient. AI will be in future even better, so if you will need to change something - talk to your Agent 🤣

**Review by pk16011990 — APPROVED** [review]
